drivers/dai/nxp/sai/sai.c

```diff
@@ -886,6 +886,21 @@ static int sai_init(const struct device *dev)
 
 	device_map(&data->regmap, cfg->regmap_phys, cfg->regmap_size, K_MEM_CACHE_NONE);
 
+	if (cfg->dline_count <= 0) {
+		LOG_ERR("Invalid number of data lines");
+		return -EINVAL;
+	}
+
+	if (cfg->tx_dline >= cfg->dline_count) {
+		LOG_ERR("Invalid TX data line index");
+		return -EINVAL;
+	}
+
+	if (cfg->rx_dline >= cfg->dline_count) {
+		LOG_ERR("Invalid RX data line index");
```

Comment:
nit: no capital letter (i.e. "invalid" vs "Invalid") to keep consistent with the other log messages